### ISSUE TITLE
list zip from a nested list ( vertices.4)

### ISSUE THREAD

**Issue by zeffii** [nortikin/sverchok#755]
for instance I have the object_in node outputting a nested list of 4 lists of vertices. Right now i'd use a scriptnode, but i'm sure there's a single node approach to this already that I haven't been smart enough to find.

https://github.com/zeffii/weblog/blob/master/blends/experimental/star_zip.blend

![image](https://cloud.githubusercontent.com/assets/619340/11685211/a05d98b2-9e77-11e5-8e9a-17c5d554cd56.png)

```python
def sv_main(verts=[[]]):
    verts_out = []
    in_sockets = [ ['v', 'some_socket_name', verts] ]

    if verts and verts[0]:
        print('len verts=', len(verts))
        print('len verts[0]=', len(verts[0]))        
        for j in range(len(verts[0])):
            for i in range(len(verts)):
                verts_out.append(verts[i][j])

    out_sockets = [ ['v', 'verts', [verts_out]] ]
    return in_sockets, out_sockets

```

if a combination of nodes is required, I propose a single `list star_zip node`

**Comment by zeffii** [comment]
is there a way to go from 
```python
vertices = [
    [ (ax, ay, az), (ax, ay, az), (ax, ay, az), (ax, ay, az) ], 
    [ (bx, by, bz), (bx, by, bz), (bx, by, bz), (bx, by, bz) ], 
    [ (cx, cy, cz), (cx, cy, cz), (cx, cy, cz), (cx, cy, cz) ]
]

to

vertices = [
    [ (ax, ay, az), (bx, by, bz), (cx, cy, cz) ], 
    [ (ax, ay, az), (bx, by, bz), (cx, cy, cz) ], 
    [ (ax, ay, az), (bx, by, bz), (cx, cy, cz) ], 
    [ (ax, ay, az), (bx, by, bz), (cx, cy, cz) ]
]

```

**Comment by ly29** [comment]
List flip does something similar to ```list(zip(* my_list))```

**Comment by zeffii** [comment]
exactly what I need..  it will be a neat to have these nodes named closer to their numpy / common implementations  names..  flatten vs unwrap.. etc.

**Comment by ly29** [comment]
A general revision of these things is needed.

**Comment by ly29** [comment]
The code in list flip is really old and also not good python code, variables named list, etc.

**Comment by ly29** [comment]
For study.
```python
def flip(list, level):
    level -= 1
    if level:
        out = []
        for l in list:
            out.extend(flip(l, level))
    else:
        out = []
        length = maxlen(list)
        for i in range(length):
            out_ = []
            for l in list:
                try:
                    out_.append(l[i])
                except:
                    out_.append(l[0])
            out.append(out_)
    return out
```
https://github.com/nortikin/sverchok/blob/master/nodes/list_interfere/flip.py#L26-L43

**Comment by zeffii** [comment]
yikes... : )

**Comment by nortikin** [comment]
it is all about recursion, folks

**Comment by ly29** [comment]
The recursion isn't the issue...

**Comment by nortikin** [comment]
try/except is bad and iterations. but if simplify recursion to avoid repeating iterations to depth f list... some kind of decorator for recursion?

**Comment by ly29** [comment]
Mostly the use of try/except and the naming the argument list that makes it bad form I guess.

Also extend instead of append makes it behave strange for deeper levels I think. Appending l[0] instead of l[-1] is not what I would expect either. But I haven't tested it so it can make sense.

But to be honest my old list nodes recursion code isn't that pretty either.

It is shame that ```list(zip( *input_list))``` doesn't work for our purposes.

Also bad python style and hard to follow code.

```python
    def match(self, lsts, level, f1, f2):
        level -= 1
        if level:
            tmp = [self.match(obj, level, f1, f2) for obj in zip(*f1(lsts))]
            return list(map(list, zip(*tmp)))
        elif type(lsts) == list:
            return f2(lsts)
        elif type(lsts) == tuple:
            return tuple(f2(list(lsts)))
        return None
```
https://github.com/nortikin/sverchok/blob/master/nodes/list_basic/match.py#L71-L80

**Comment by ly29** [comment]
A place where the naming the argument list causes problems.
```
#  consider this.
...
   elif type(list) in [type([])]:
...
```

```python
    def revers(self, list, level):
        level -= 1
        if level:
            out = []
            for l in list:
                out.append(self.revers(l, level))
            return out
        elif type(list) in [type([])]:
            return list[::-1]
        elif type(list) in [type(tuple())]:
            return list[::-1]
```

**Comment by zeffii** [comment]
@nortikin I don't really care how code is written if it's fast or works well. Sometimes it's nice to find code written this way because most likely improvements can be made.

**Comment by nortikin** [comment]
it is happened if non-programmer try to develop something. Python so flexible so you can make much of this mistakes

**Comment by ly29** [comment]
yeah, I sure do.
